app/assets/javascripts/feed.js

```diff
@@ -0,0 +1,31 @@
+"use strict";
+function init_users_feed_page() {
+    $(window).unload(function () {
+        var readed_events = JSON.stringify(get_readed_events_ids());
```

Comment:
I don't like that you executing this JS every time when page is "unloaded"
you can do it in controller when you loading 3 items, plus what if request will not be executed, items still be be in "unread" state